Bootstrap/Update.php

```diff
@@ -466,4 +467,47 @@ private function createRemoteToLocalCategoriesTable()
             }
         }
     }
+
+    private function recreateRemoteCategoriesAndProductAssignments()
+    {
+        if (version_compare($this->version, '1.1.4', '<=')) {
+            try {
+                $result = $this->db->query('SELECT `article_id`, `category` FROM `s_plugin_connect_items` WHERE shop_id IS NOT NULL');
+
+                while ($row = $result->fetch()) {
+                    $categories = json_decode($row['category'], true);
+                    $countAssignedCategories = $this->db->query('SELECT COUNT(`connect_category_id`) AS categories_count FROM s_plugin_connect_product_to_categories WHERE articleID = ?',
+                        [$row['article_id']]
+                    )->fetch();
+
+                    if (count($categories) != $countAssignedCategories['categoriesCount']) {
```

Comment:
this doesn't work. used key in query is not categoriesCount, it is categories_count
btw better use ```$countAssignedCategories = $this->db->query('SELECT COUNT(`connect_category_id`) FROM s_plugin_connect_product_to_categories WHERE articleID = ?',
                        [$row['article_id']]
                    )->fetchColumn();```. then ```$countAssignedCategories``` contains result of your count.